npm_and_yarn/lib/dependabot/npm_and_yarn/update_checker/registry_finder.rb

```diff
@@ -49,16 +49,22 @@ def self.central_registry?(registry)
         end
 
         def registry_from_rc(dependency_name)
-          return global_registry unless dependency_name.start_with?("@") && dependency_name.include?("/")
-
-          scope = dependency_name.split("/").first
-          scoped_registry(scope)
+          explicit_registry_from_rc(dependency_name) || global_registry
```

Comment:
I did this for backwards compatibility with other usages of `RegistryFinder#registry_from_rc`. I think it's used just once and could probably use the changed behavior too, I'll look into what would mean to change it.

It would make the code a bit simpler and may fix additional issues, so worth looking into it.

Reply:
I had a look at this but decided it was not worth exploring further right now.